releasenotes/notes/46016.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: traffic-management
+issue:
+  - 45335
+releaseNotes:
+  - |
+    **Fixed** an issue preventing dualstack support for EKS IPv6 clusters with secondary IPv4 link-local address.
+
+upgradeNotes:
+  - title: Use the excluded interface option for Ignoring IPv4 in an IPv6 cluster.
+    content: |
+      If you don't wish to use Istio dualstack feature, you will need to add v4if0 to excluded interface list. e.g. traffic.sidecar.istio.io/excludeInterfaces: v4if0
```

Comment:
this seems like a breaking change. in a single stack ipv6 env I see readiness probes failing now (EKS pods listen on `0.0.0.0` private ipv4 rather than `::` as they did before)

applying this annotation does fix this. but i thought dual stack was feature gated and still experimental. should this be feature gated?

cc @howardjohn @ilrudie

Reply:
+1 for feature gated

Reply:
the problem right now is that link-local address is hard-coded and completely ignored. i understand that users have to update exclude interface which is more or less the same as users are doing today by excluding 0.0.0.0 in the exclude ip range. and exclude interface doesn't turned by DS. if user want DS, that makes the excludeinterface optional and user still have to apply the DS profile to enable it.